SU2_CFD/include/fluid/CFluidScalar.hpp

```diff
@@ -0,0 +1,113 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+#include <vector>
+#include <memory>
+
+#include "CFluidModel.hpp"
+
+/*!
+ * \class CFluidScalar
+ * \brief Child class for defining an incompressible ideal gas model.
+ * \author: T. Economon
+ */
+class CFluidScalar final : public CFluidModel {
+private:
+  unsigned short n_scalars = 0;                       /*!< \brief Number of transported scalars. */
+  unsigned short n_species_mixture = 0;               /*!< \brief Number of species in mixture. */
+  su2double Gas_Constant = 0.0;                       /*!< \brief Specific gas constant. */
+  su2double Gamma = 0.0;                              /*!< \brief Ratio of specific heats of the gas. */
+  su2double Pressure_Thermodynamic = 0.0;             /*!< \brief Constant pressure thermodynamic. */
+
+  bool wilke;
+  bool davidson;
+
+  std::vector<su2double> massFractions;               /*!< \brief Mass fractions of all species. */
+  std::vector<su2double> moleFractions;               /*!< \brief Mole fractions of all species. */
+  std::vector<su2double> molarMasses;                 /*!< \brief Molar masses of all species. */
+  std::vector<su2double> laminarViscosity;            /*!< \brief Laminar viscosity of all species. */
+  std::vector<su2double> specificHeat;                /*!< \brief Specific heat of all species. */
+  std::vector<su2double> laminarthermalConductivity;  /*!< \brief Laminar thermal conductivity of all species. */
+
+  static const int ARRAYSIZE = 100;
+  std::unique_ptr<CViscosityModel> LaminarViscosityPointers[ARRAYSIZE];
+  std::unique_ptr<CConductivityModel> ThermalConductivityPointers[ARRAYSIZE];
+
+ public:
+  /*!
+   * \brief Constructor of the class.
+   */
+  CFluidScalar(su2double val_Cp, su2double val_gas_constant, su2double val_operating_pressure, CConfig *config);
+  /*!
+   * \brief Set viscosity model.
+   */
+  void SetLaminarViscosityModel(const CConfig* config) override;
+
+  /*!
+   * \brief Set thermal conductivity model.
+   */
+  void SetThermalConductivityModel(const CConfig* config) override;
+  
+  /*!
+   * \brief Set the Dimensionless State using Temperature.
+   * \param[in] t - Temperature value at the point.
+   */
+  void SetTDState_T(su2double t){
+    /*--- The EoS only depends upon temperature. ---*/
+    Temperature = t;
+    Density = Pressure / (Temperature * Gas_Constant);
+  }
+  /*!
+   * \brief Set the Dimensionless State using Temperature.
+   * \param[in] t - Temperature value at the point.
+   */
+
+  void SetTDState_T(const su2double val_temperature, su2double * const val_scalars);
+
+   /*!
+   * \brief Get fluid dynamic viscosity.
+   */
+  virtual inline su2double GetLaminarViscosity() {
```

Comment:
is it supposed to virtual or override ? check the other methods please

Reply:
I just checked this with Cristopher and the c++ guideline says that the base class should be virtual and the derived class override, and each virtual function declaration should specify either virtual,override or final exactly once.
So we follow this advice and make this override
http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines.html#c128-virtual-functions-should-specify-exactly-one-of-virtual-override-or-final